### ISSUE TITLE
Add support for MSK IAM authentication in `rdkafka2` output type

### ISSUE THREAD

**Issue by madebydna** [fluent/fluent-plugin-kafka#522]
### Is your feature request related to a problem? Please describe.

AWS MSK requires a signed token for their IAM based authentication with OAUTHBEARER. This is currently not supported by the plugin so there is no way to authenticate FluentD to send messages to an MSK cluster. 

Previously, [a similar suggestion was made](https://github.com/fluent/fluent-plugin-kafka/issues/472) but in the context of the `ruby-kafka` library which is deprecated. This feature could, however, be supported using the `rdkafka2` output type. 

### Describe the solution you'd like

I created a [fork](https://github.com/madebydna/fluent-plugin-kafka) that adds basic support for MSK IAM authentication with the help of the [aws-msk-iam-sasl-signer-ruby](https://github.com/bruce-szalwinski-he/aws-msk-iam-sasl-signer-ruby) library. I essentially used the [example code](https://github.com/bruce-szalwinski-he/aws-msk-iam-sasl-signer-ruby/tree/main/examples/rdkafka) for the token generation when using rdkafka as the underlying Kafka library. 

However, I'm not sure if this approach is solid enough, so I was wondering if someone could rework the code (or suggest improvements) so that it could be merged into the official plugin. For instance, my solution requires using the shared producer (config `share_producer true`) which is not the default. I wasn't able to use the multi-threaded producer feature as it led to a `ThreadError` due to recursive locking.

### Describe alternatives you've considered

Since I need this feature for use in our production environment, I'm planning on using my own fork for the time being. However, I would be more comfortable if this feature was officially supported and did not require using a modified fork of this plugin.

**Comment by Watson1978** [comment]
#521 is the PR for this issue?

**Comment by daipom** [comment]
If you could leave it open for now, it would be easier for someone familiar with it to advise you.

**Comment by madebydna** [comment]
Yes, that is the PR. I actually accidentally opened it, as I wanted to ask general advice beforehand, but I can certainly reopen it again if that would make things easier.
